**Importing necessary libraries**

In [79]:
import pandas as pd

**Data Loading**

In [80]:
df = pd.read_csv('/content/Titanic-Dataset.csv')

**Identifying the missing Values**

In [81]:
missing_count = df.isnull().sum()
missing_count

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,177
SibSp,0
Parch,0
Ticket,0
Fare,0


**Handling Missing Values**

Cabin has too many missing values, so we drop it

In [82]:
df = df.drop(columns=['Cabin'])

In [ ]:
# df.head(10)

 Age has 177 missing values so fill them with mean to preserve data


In [83]:
df['Age'] = df['Age'].fillna(df['Age'].mean())

In [84]:
missing_count = df.Age.isnull().sum()
missing_count

np.int64(0)

**Fix Wrong DataType**

Although the age column was filled but it retained a float datatype due to previous NAN values.

In [85]:
df.Age.dtype

dtype('float64')

In [86]:
# Convert the age column to integer type
df['Age'] = df['Age'].astype(int)


**Remove Duplicates**

In [87]:
total_duplicates = df.duplicated().sum()
print(total_duplicates)

0


 No duplicates fount, so we dont need to run drop_duplicates()

**Create 2 New Derived Columns**

In [88]:
# Combineing SibSp and Parsch helps analyze the total family size
df['FamilySize'] = df['SibSp']+df['Parch']+1

In [89]:
# Categorizing age helps simplifying analysis.
df['AgeGroup'] = df['Age'].apply(lambda x: 'Child' if x<18 else 'Adult' )

**Reshape Data (Summary Table)**

Aggreagated data by 'Pclass' to analyze the count of passengers and their average age, highlighting demographic patterns across class

In [90]:
# Creating a summary table to see survival rates by class
summary_table = df.groupby('Pclass').agg({'Survived': 'count', 'Age':'mean',})
summary_table

,Survived,Age
Pclass,,
1,216,36.944444
2,184,29.804348
3,491,26.191446


**Mapping and Merging**

Used a left join to map numerical 'Pclass' values to descriptive labels ('First', 'Second', 'Third') while preserving all original dat

In [91]:
#  Create mapping
class_mapping = pd.DataFrame({'Pclass': [1,2,3], 'ClassName': ['First', 'Second', 'Third']})

# Merge to add the label

df = df.merge(class_mapping, on='Pclass', how = 'left')



In [92]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,FamilySize,AgeGroup,ClassName
0,1,0,3,"Braund, Mr. Owen Harris",male,22,1,0,A/5 21171,7.2500,S,2,Adult,Third
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38,1,0,PC 17599,71.2833,C,2,Adult,First
2,3,1,3,"Heikkinen, Miss. Laina",female,26,0,0,STON/O2. 3101282,7.9250,S,1,Adult,Third
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35,1,0,113803,53.1000,S,2,Adult,First
4,5,0,3,"Allen, Mr. William Henry",male,35,0,0,373450,8.0500,S,1,Adult,Third
